hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -220,6 +269,8 @@ private void initConf(HiveConf conf, int port) {
     conf.set(HiveConf.ConfVars.METASTORE_TRY_DIRECT_SQL.varname, "false");
     conf.set(HiveConf.ConfVars.METASTORE_DISALLOW_INCOMPATIBLE_COL_TYPE_CHANGES.varname, "false");
     conf.set("iceberg.hive.client-pool-size", "2");
+    // hive 1.x requires a valid DN connection pool, use bonecp as it exposes a close method
+    conf.set("datanucleus.connectionPoolingType", "BONECP");
```

Comment:
Would this allow the tests to run with Hive 1.x as well?

Reply:
I used gradle `dependencies` and checked `testRuntimeClasspath` of spark 2.4. And the hive version is `org.spark-project.hive:hive-exec:1.2.1.spark2`. So I suppose we're already running with 1.x

Reply:
Why did we not face this issue before? What is new?

Reply:
It's not an issue. BONECP is the default value. I just explicitly set it here in case the default value changes in the future.

Reply:
I would just keep the default if it does not hurt